runtime/doc/autocmd.txt

```diff
@@ -778,11 +778,10 @@ OptionSet			After setting an option (except during
 				This does not set |<abuf>|, you could use
 				|bufnr()|.
 
-				Note that when setting a |global-local| string
-				option with |:set|, then |v:option_old| is the
-				old global value. However, for all other kinds
-				of options (local string options, global-local
-				number options, ...) it is the old local
+				Note that when setting a |global-local| option
```

Comment:
I'd add a brief note in the commit message body about this, as it is a functional change.